src/web/app/pages-session/session-submission-page/session-submission-page.component.spec.ts

```diff
@@ -1,24 +1,572 @@
+import { FeedbackResponsesService } from '../../../services/feedback-responses.service';
+import { FeedbackSessionsService } from '../../../services/feedback-sessions.service';
+import { InstructorService } from '../../../services/instructor.service';
+import { NavigationService } from '../../../services/navigation.service';
+import { SimpleModalService } from '../../../services/simple-modal.service';
+import { StudentService } from '../../../services/student.service';
+import {
+  AuthInfo,
+  CommentVisibilityType,
+  FeedbackConstantSumQuestionDetails,
+  FeedbackConstantSumResponseDetails,
+  FeedbackContributionQuestionDetails,
+  FeedbackContributionResponseDetails,
+  FeedbackMcqQuestionDetails,
+  FeedbackMcqResponseDetails,
+  FeedbackMsqQuestionDetails,
+  FeedbackMsqResponseDetails,
+  FeedbackNumericalScaleQuestionDetails,
+  FeedbackNumericalScaleResponseDetails,
+  FeedbackParticipantType,
+  FeedbackQuestionRecipients,
+  FeedbackQuestionType,
+  FeedbackRankOptionsQuestionDetails,
+  FeedbackRankOptionsResponseDetails,
+  FeedbackRankRecipientsQuestionDetails,
+  FeedbackResponse,
+  FeedbackResponseComment,
+  FeedbackRubricQuestionDetails,
+  FeedbackRubricResponseDetails,
+  FeedbackSession,
+  FeedbackSessionPublishStatus,
+  FeedbackSessionSubmissionStatus,
+  FeedbackTextQuestionDetails,
+  FeedbackTextResponseDetails,
+  FeedbackVisibilityType,
+  Instructor,
+  JoinState,
+  NumberOfEntitiesToGiveFeedbackToSetting,
+  RegkeyValidity,
+  ResponseVisibleSetting,
+  SessionVisibleSetting,
+  Student,
+} from '../../../types/api-output';
+import { Intent } from '../../../types/api-request';
 import { AjaxLoadingModule } from '../../components/ajax-loading/ajax-loading.module';
+import { CommentRowModel } from '../../components/comment-box/comment-row/comment-row.component';
 import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
 import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
+import {
+  FeedbackResponseRecipientSubmissionFormModel,
+  QuestionSubmissionFormModel,
+} from '../../components/question-submission-form/question-submission-form-model';
 import {
   QuestionSubmissionFormModule,
 } from '../../components/question-submission-form/question-submission-form.module';
 import { TeammatesCommonModule } from '../../components/teammates-common/teammates-common.module';
-import { SessionSubmissionPageComponent } from './session-submission-page.component';
+import { SavingCompleteModalComponent } from './saving-complete-modal/saving-complete-modal.component';
+import { FeedbackResponsesResponse, SessionSubmissionPageComponent } from './session-submission-page.component';
+import Spy = jasmine.Spy;
 
 describe('SessionSubmissionPageComponent', () => {
+  const deepCopy: any = (obj: any) => JSON.parse(JSON.stringify(obj));
+
+  const testStudent: Student = {
```

Comment:
I think it can be a lot cleaner if we put these sample data into a separate file

Reply:
Also it might be better to use some kind of logic to generate the data instead of hard-coded data

Reply:
Perhaps it's just my preference but I think it's ok to leave it as it is? There's not too much of overlap in the fields of each item, so might as well state it as it is, plus extracting it out into logic can make it harder to read

Reply:
hmm I feel like sample data is taking a significant LoC in this file, thats why I suggested separating them or compacting them by using some kind of loop

Reply:
Agree that the data takes up a lot of space, it's possible to separate test data into a separate file (as with `services`), though for the main components and pages there looks to be no precedent for this
Alternatively we could also remove some of the types of questions and pick just a few to make it shorter?